pkg/gofr/datasource/solr/solr.go

```diff
@@ -0,0 +1,215 @@
+
+	return call(ctx, "POST", url, params, document)
+}
+
+// ListFields retrieves all the fields in the schema for the specified collection.
+// params can be used to send query parameters like wt, fl, includeDynamic etc.
+func (c *Client) ListFields(ctx context.Context, collection string, params map[string]any) (any, error) {
+	url := c.url + collection + "/schema/fields"
+
+	defer c.sendOperationStats(&QueryLog{Type: "ListFields", Url: url}, time.Now())
+
+	return call(ctx, "GET", url, params, nil)
+}
+
+// Retrieve retrieves the entire schema that includes all the fields,field types,dynamic rules and copy field rules.
+// params can be used to specify the format of response
+func (c *Client) Retrieve(ctx context.Context, collection string, params map[string]any) (any, error) {
+	url := c.url + collection + "/schema"
+
+	defer c.sendOperationStats(&QueryLog{Type: "Retrieve", Url: url}, time.Now())
+
+	return call(ctx, "GET", url, params, nil)
+}
+
+// AddField adds Field in the schema for the specified collection
+func (c *Client) AddField(ctx context.Context, collection string, document *bytes.Buffer) (any, error) {
+	url := c.url + collection + "/schema"
+
+	defer c.sendOperationStats(&QueryLog{Type: "AddField", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, nil, document)
+}
+
+// UpdateField updates the field definitions in the schema for the specified collection
+func (c *Client) UpdateField(ctx context.Context, collection string, document *bytes.Buffer) (any, error) {
+	url := c.url + collection + "/schema"
+
+	defer c.sendOperationStats(&QueryLog{Type: "UpdateField", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, nil, document)
+}
+
+// DeleteField deletes the field definitions in the schema for the specified collection
+func (c *Client) DeleteField(ctx context.Context, collection string, document *bytes.Buffer) (any, error) {
+	url := c.url + collection + "/schema"
+
+	defer c.sendOperationStats(&QueryLog{Type: "DeleteField", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, nil, document)
+}
+
+// Response stores the response from SOLR
+type Response struct {
+	Code int
+	Data any
+}
+
+// call forms the http request and makes a call to solr and populates the solr response
+func call(ctx context.Context, method, url string, params map[string]any, body io.Reader) (any, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	if method != "GET" {
+		req.Header.Add("content-type", "application/json")
+	}
+
+	q := req.URL.Query()
+
+	for k, val := range params {
+		switch v := val.(type) {
+		case []string:
+			for _, val := range v {
+				q.Add(k, val)
+			}
+		default:
+			q.Add(k, fmt.Sprint(val))
+		}
+	}
+
+	req.URL.RawQuery = q.Encode()
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req.WithContext(ctx))
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var respBody any
+
+	b, _ := io.ReadAll(resp.Body)
```

Comment:
Why ignoring the error?


```suggestion
	b, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
```